src/buildReader.ts

```diff
@@ -327,8 +327,9 @@ export default function buildReader(types: readonly RosMsgDefinition[]): Seriali
         // Typed arrays are considered native types and remain as typed arrays
         toJSON() {
           const view = this.#view;
-          const buffer = new Uint8Array(view.buffer, view.byteOffset, view.byteLength);
-          return fullReader.readMessage(buffer);
+          const buffer = new Uint8Array(view.buffer, view.byteOffset + this.#offset, view.byteLength - this.#offset);
+
+          return typeReaders.get("${name}").readMessage(buffer);
```

Comment:
nit:
```suggestion
          return typeReaders.get(${JSON.stringify(name)}).readMessage(buffer);
```

Reply:
What's the benefit of stringifying the name? We know it can't have strings in it per the ROS spec.

Reply:
But what if it did anyway? :philosoraptor: